app/models/dashboard.rb

```diff
@@ -16,7 +16,7 @@ class Dashboard < ActiveRecord::Base
   belongs_to :directory
   has_many :shortened_urls
 
-  validates :name, uniqueness: { case_sensitive: false }
+  validates :name, :slug, uniqueness: { case_sensitive: false }
```

Comment:
While this is great first step to validate uniqueness, the only way to really ensure it is to have a unique constraint on the database. Otherwise it's possible that in case of race conditions (two requests get executed at the same time) we still end up with duplicates.